src/ui/units/collections/components/CollectionContent/CollectionContent.tsx

```diff
@@ -152,16 +212,257 @@ export const CollectionContent = React.memo<Props>(
+                                        if (collection) {
+                                            dispatch(updateCollectionInItems(collection));
+                                        }
+                                    },
+                                    onClose: () => {
+                                        dispatch(closeDialog());
+                                    },
+                                },
+                            }),
+                        );
+                    },
+                });
+            }
+
+            if (item.permissions.move) {
+                actions.push({
+                    text: i18n('action_move'),
+                    action: () => {
+                        dispatch(
+                            openDialog({
+                                id: DIALOG_MOVE_COLLECTION,
+                                props: {
+                                    open: true,
+                                    collectionId: item.collectionId,
+                                    collectionTitle: item.title,
+                                    initialParentId: item.parentId,
+                                    onApply: refreshContent,
+                                    onClose: handeCloseMoveDialog,
+                                },
+                            }),
+                        );
+                    },
+                });
+            }
+
+            if (collectionsAccessEnabled && item.permissions.listAccessBindings) {
+                actions.push({
+                    text: i18n('action_access'),
+                    action: () => {
+                        setDialogState(DialogState.EditCollectionAccess);
+                        setDialogEntity(item);
+                    },
+                });
+            }
+
+            if (item.permissions.delete) {
+                actions.push({
+                    text: i18n('action_delete'),
+                    action: () => {
+                        setDialogState(DialogState.DeleteCollection);
+                        setDialogEntity(item);
+                    },
+                    theme: 'danger',
+                });
+            }
+
+            return actions;
+        };
+
         return (
             <React.Fragment>
-                <CollectionContentTable
-                    contentItems={contentItems}
-                    filters={filters}
-                    setFilters={setFilters}
-                    refreshContent={refreshContent}
-                />
+                {filters.viewMode === ViewMode.Grid ? (
+                    <CollectionContentGrid
+                        contentItems={contentItems}
+                        filters={filters}
+                        setFilters={setFilters}
+                        getWorkbookActions={getWorkbookActions}
+                        getCollectionActions={getCollectionActions}
+                        onClickStopPropagation={onClickStopPropagation}
```

Comment:
`onClickStopPropagation` is just a static method and can be placed in `utils.ts` and imported in components.